### ISSUE TITLE
I can't open settings

### ISSUE THREAD

**Issue by JWOverschot** [dsafa/audio-band#108]
When I try to open the settings It doesn't do anything.
The option is not disabled and is clickable but doesn't open the settings.

I have checked if I had .Net framework 4.7 as described in documentation.
But it was already installed.

OS: Windows 10 1809.

**Comment by jgleesti** [comment]
I'm having the same issue
OS: Windows 10 1803

**Comment by JWOverschot** [comment]
I have tried installing it on two other computers and I have the same issue.

**Comment by dsafa** [comment]
Attach the logs from %temp%\AudioBand.log and I'll take a look

**Comment by jgleesti** [comment]
I cleared the logs from before and re-installed to find exactly when the logs were generated. They were generated when the toolbar first loads into the taskbar. No additional logs are created when the "Audio Band Settings" menu item is clicked.

2019-03-06 20:25:17.8690|DEBUG|Audio Band|Initialization started
2019-03-06 20:25:19.5069|DEBUG|AudioBand.ViewModels.CustomLabelVM|Starting edit
2019-03-06 20:25:19.6029|DEBUG|AudioBand.ViewModels.AlbumArtPopupVM|Starting edit
2019-03-06 20:25:19.6029|DEBUG|AudioBand.ViewModels.AlbumArtVM|Starting edit
2019-03-06 20:25:19.6346|DEBUG|AudioBand.ViewModels.AudioBandVM|Starting edit
2019-03-06 20:25:19.8116|DEBUG|AudioBand.ViewModels.NextButtonVM|Starting edit
2019-03-06 20:25:19.8386|DEBUG|AudioBand.ViewModels.PlayPauseButtonVM|Starting edit
2019-03-06 20:25:19.8626|DEBUG|AudioBand.ViewModels.PreviousButtonVM|Starting edit
2019-03-06 20:25:19.8856|DEBUG|AudioBand.ViewModels.ProgressBarVM|Starting edit
2019-03-06 20:25:20.4971|ERROR|Audio Band|Error during initialization
---START OF EXCEPTION---
System.Windows.Markup.XamlParseException: Could not load file or assembly 'MahApps.Metro.IconPacks.Material, PublicKeyToken=null' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)
   at void System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)
   at object System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, bool skipJournaledProperties, object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
   at object System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, bool skipJournaledProperties, object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   at object System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, object parent, bool closeStream)
   at new AudioBand.Views.Wpf.SettingsWindow(SettingsWindowVM vm) in D:/a/1/s/src/AudioBand/Views/Wpf/SettingsWindow.xaml.cs:line 37
   at Task AudioBand.MainControl.InitializeAsync()+() => { } in D:/a/1/s/src/AudioBand/MainControl.cs:line 133
   at object System.Windows.Threading.DispatcherOperation.InvokeDelegateCore()
   at void System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at async Task AudioBand.MainControl.InitializeAsync() in D:/a/1/s/src/AudioBand/MainControl.cs:line 139 ---> System.IO.FileLoadException: Could not load file or assembly 'file:///C:\audioband\MahApps.Metro.IconPacks.Material.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)
   at RuntimeAssembly System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, string codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, ref StackCrawlMark stackMark, IntPtr pPrivHostBinder, bool throwOnFileNotFound, bool forIntrospection, bool suppressSecurityChecks)
   at RuntimeAssembly System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, ref StackCrawlMark stackMark, IntPtr pPrivHostBinder, bool throwOnFileNotFound, bool forIntrospection, bool suppressSecurityChecks)
   at RuntimeAssembly System.Reflection.RuntimeAssembly.InternalLoadFrom(string assemblyFile, Evidence securityEvidence, byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, bool forIntrospection, bool suppressSecurityChecks, ref StackCrawlMark stackMark)
   at Assembly System.Reflection.Assembly.LoadFrom(string assemblyFile)
   at Assembly AudioBand.Views.Wpf.SettingsWindow.CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args) in D:/a/1/s/src/AudioBand/Views/Wpf/SettingsWindow.xaml.cs:line 90
   at RuntimeAssembly AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, string assemblyFullName) ---> System.NotSupportedException: An attempt was made to load an assembly from a network location which would have caused the assembly to be sandboxed in previous versions of the .NET Framework. This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous. If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch. See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.

   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
---END OF EXCEPTION---
---ORIGINAL EXCEPTION---
System.Windows.Markup.XamlParseException: Could not load file or assembly 'MahApps.Metro.IconPacks.Material, PublicKeyToken=null' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515) ---> System.IO.FileLoadException: Could not load file or assembly 'file:///C:\audioband\MahApps.Metro.IconPacks.Material.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515) ---> System.NotSupportedException: An attempt was made to load an assembly from a network location which would have caused the assembly to be sandboxed in previous versions of the .NET Framework. This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous. If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch. See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.
   --- End of inner exception stack trace ---
   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at AudioBand.Views.Wpf.SettingsWindow.CurrentDomain_AssemblyResolve(Object sender, ResolveEventArgs args) in D:\a\1\s\src\AudioBand\Views\Wpf\SettingsWindow.xaml.cs:line 90
   at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
   --- End of inner exception stack trace ---
   at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
   at AudioBand.Views.Wpf.SettingsWindow..ctor(SettingsWindowVM vm) in D:\a\1\s\src\AudioBand\Views\Wpf\SettingsWindow.xaml.cs:line 37
   at AudioBand.MainControl.<InitializeAsync>b__35_1() in D:\a\1\s\src\AudioBand\MainControl.cs:line 133
   at System.Windows.Threading.DispatcherOperation.InvokeDelegateCore()
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AudioBand.MainControl.<InitializeAsync>d__35.MoveNext() in D:\a\1\s\src\AudioBand\MainControl.cs:line 139
---END OF ORIGINAL EXCEPTION

**Comment by dsafa** [comment]
Unblock the file called `MahApps.Metro.IconPacks.Material`. You may have to unblock others too but that is the one causing issues.

**Comment by JWOverschot** [comment]
@dsafa Thanks, I just unblocked the whole zip and now it works.

**Comment by jgleesti** [comment]
That does not appear to have worked for me. I unblocked all files in the directory and ran the unblock.ps1 script. I also uninstalled and reinstalled the toolbar. The same error log with one minor difference appears to be generated when the toolbar is added to the taskbar. From what I can see the difference is 'MahApps.Metro.IconPacks.Material, PublicKeyToken=null' changed to 'file:///C:\audioband\MahApps.Metro.IconPacks.Material.dll'
[AudioBand.log](https://github.com/dsafa/audio-band/files/2941739/AudioBand.log)
